test/test_read_all.py

```diff
@@ -72,12 +78,22 @@ def read_with_zarrita(fpath, ds_name):
     h = zarrita.get_hierarchy(str(fpath.absolute()))
     return h["/" + ds_name][:]
 
+def read_with_xtensor_zarr(fpath, ds_name):
+    if ds_name == "blosc":
+        ds_name = "blosc/lz4"
+    fname = "a.npz"
+    if os.path.exists(fname):
+        os.remove(fname)
+    subprocess.check_call(["generate_data/xtensor_zarr/build/run_xtensor_zarr", fpath, ds_name])
+    return np.load(fname)["a"]
+
```

Comment:
This is a nice approach. I see that you built this .npz writer into the `main.cpp` program whenever two command line arguments are provided.

Reply:
Yes, we use the same executable for writing and reading.